recipes/openssl/1.x.x/conanfile.py

```diff
@@ -223,35 +216,61 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
-
-    def layout(self):
-        pass
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+        self.conf.define("tools.gnu:make_program", self._make_program)
 
     def requirements(self):
-        if self._full_version < "1.1.0" and self.options.get_safe("no_zlib") == False:
+        if self._full_version < "1.1.0" and not self.options.get_safe("no_zlib"):
             self.requires("zlib/1.2.12")
 
     def validate(self):
-        if self.settings.os == "Emscripten":
+        if self.info.settings.os == "Emscripten":
             if not all((self.options.no_asm, self.options.no_threads, self.options.no_stdio, self.options.no_tests)):
                 raise ConanInvalidConfiguration("os=Emscripten requires openssl:{no_asm,no_threads,no_stdio,no_tests}=True")
 
     def build_requirements(self):
         if self._settings_build.os == "Windows":
-            if not self._win_bash:
-                self.build_requires("strawberryperl/5.30.0.1")
-            if not self.options.no_asm and not tools.which("nasm"):
-                self.build_requires("nasm/2.15.05")
-        if self._win_bash and not tools.get_env("CONAN_BASH_PATH"):
+            if not self.win_bash:
+                self.tool_requires("strawberryperl/5.30.0.1")
+            if not self.options.no_asm and not shutil.which("nasm"):
```

Comment:
should we always use the one from ConanCenter? Let the user override with a system version if they like?

Reply:
Hi @prince-chrismc,
Please see the comment at lin 679, which predates my changes. It reads:
>            # enforce strawberry perl, otherwise wrong perl could be used (from Git bash, MSYS, etc.)
Thus I'm inclined to recommend leaving it as is for now. Let me know if you agree. Thanks

Reply:
Sorry it was not clear in my comment but this was specifically related to `nasm` there's a reason why `tools.which` is deprecated and I wanted to double check with the rest of the team if we wanted to keep that functionality.

We'd like to removing this going forward, like build tools, these should be pulled from ConanCenter and end consumers can use the override mechanism to change that if they so wish.

What we'd like to see is

```suggestion
            if not self.options.no_asm:
```